### PR TITLE
Point cloud UX improvements

### PR DESCRIPTION
**Pull request by nyalldawson**
A bunch of UX improvements, and general nicer handling of things via the provider registry instead of hardcoded logic.

Including:

- Only expose a single generic "Point Cloud" tab in data source manager, and use provider registry methods to auto select the appropriate provider to use for a file
- Hide a bunch of unwanted files from browser, such as the broken ogr entries for "ept.json" files and the internal "ept-build.json" files.

### CODE REVIEW

**Comment by nyalldawson** [comment]
Ping @PeterPetrik

File: cmake/FindPDAL.cmake [hunk 1]
```diff
@@ -17,28 +17,26 @@
...
+
 FIND_PATH(PDAL_INCLUDE_DIR pdal.hpp
-  "$ENV{LIB_DIR}/include/"
+  "$ENV{LIB_DIR}/include"
   "$ENV{LIB_DIR}/include/pdal"
+  "/usr/include/pdal"
   c:/msys/local/include
   NO_DEFAULT_PATH
   )
-FIND_PATH(PDAL_INCLUDE_DIR pdal.hpp)
```
Comment (PeterPetrik) [inline cmake/FindPDAL.cmake:26]: I would keep it, see comment on lines 14-19. (we have the same in other Find* files)

File: src/core/qgsfileutils.h [hunk 2]
```diff
@@ -46,6 +46,14 @@ class CORE_EXPORT QgsFileUtils
      */
     static QStringList extensionsFromFilter( const QString &filter );
 
+    /**
+     * Giiven a \a filter string like "GeoTIFF Files (*.tiff *.tif)", extracts
```
Comment (PeterPetrik) [inline src/core/qgsfileutils.h:50]: Given

**Review by PeterPetrik — APPROVED** [review]
I suggest to merge and revert the FindPDAL on master?

File: cmake/FindPDAL.cmake (hunk 1 above)
Comment (nyalldawson) [reply cmake/FindPDAL.cmake]: Ok, reverted

File: src/core/qgsfileutils.h (hunk 2 above)
Comment (nyalldawson) [reply src/core/qgsfileutils.h]: FIxed

**Pull request merged**